tensorflow/core/lib/strings/strcat.h

```diff
@@ -229,6 +230,10 @@ inline void StrAppend(string *dest, const AlphaNum &a, const AlphaNum &b,
                          {a.Piece(), b.Piece(), c.Piece(), d.Piece(), e.Piece(),
                           static_cast<const AlphaNum &>(args).Piece()...});
 }
+template <class T>
+inline void StrAppend(string *dest, const std::complex<T> &a) {
```

Comment:
I think it probably makes sense to add support to AlphaNum to support std::complex rather than add a separate 1 argument StrAppend special-case.

Reply:
I saw this option as well. It is slightly more invasive to implement it this way but you are right, that may be cleaner. I will re-work it.

Reply:
My view is that at some point in the future, TF might switch to using https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_cat.h, and it might make sense to keep things in line with how they might do things.

In fact, it might be worth sending either a PR or filing an issue to make sure that these two libraries stay roughly in sync.